ddraw/IDirect3DDeviceX.cpp

```diff
@@ -1431,21 +1456,31 @@ HRESULT m_IDirect3DDeviceX::BeginScene()
 		// Set 3D Enabled
 		ddrawParent->Enable3D();
 
-		return (*d3d9Device)->BeginScene();
+		HRESULT hr = (*d3d9Device)->BeginScene();
+
+#if WITH_IMGUI
+		ImGui_ImplDX9_NewFrame();
```

Comment:
This block should probably check `ShowDebugUI` too, which would imply that static variable has to be moved to a scope both references can access